src/hotspot/cpu/x86/stubGenerator_x86_64_tanh.cpp

```diff
@@ -324,23 +324,25 @@ address StubGenerator::generate_libmTanh() {
   __ enter(); // required for proper stackwalking of RuntimeStub frame
 
   __ bind(B1_2);
+  __ pextrw(rcx, xmm0, 3);
+  __ movl(rdx, 32768);
+  __ andl(rdx, rcx);
+  __ andl(rcx, 32767);
+  __ cmpl(rcx, 16436);
+  __ jcc(Assembler::aboveEqual, L_2TAG_PACKET_2_0_1); // Branch only if |x| >= 20
```

Comment:
It will be good to return +/- 1 when |x| >= 22 instead of 20 to match the Java code.

Reply:
Yes, I made the change. Thanks.